fluent-react/test/provider_valid.test.js

```diff
@@ -25,18 +25,4 @@ describe("LocalizationProvider - validation", () => {
     );
     expect(renderer.toJSON()).toMatchInlineSnapshot(`<div />`);
   });
-
-  test("without a child", () => {
```

Comment:
Suggestion (blocking): I'm nervous about this change, since _we_ can't do this with TypeScript type checks, but our users might. For instance in Firefox DevTools this is used without type checks.

Reply:
This PR drops two tests. They expected these to throw:

```jsx
<LocalizationProvider l10n={new ReactLocalization([])} />
```
```jsx
<LocalizationProvider />
```

I don't think there's any actual benefit from throwing for these, and doing so is in fact surprising. A `LocalizationProvider` is a [`Context.Provider`](https://reactjs.org/docs/context.html#contextprovider), which normally does not _require_ its children to be non-empty. It just does nothing in that case.

The second checks that the `l10n` prop is defined. This will continue to throw in actual use, as the users of the context get its value with the TS type `ReactLocalization | null`, and handle the latter by throwing. This is already effectively checked by tests in `fluent-react/test/localized_valid.test.js`.

@gregtatum Would you like me to add a more specific test to the `Localized` test suite covering the missing-prop case?

Reply:
I think there is some value in exercising these code paths as users of this package could do surprising things with it. Without the throws, I'm less opinionated if the tests stay or if they go.